Automattic/pocket-casts-android
# Remove GlobalScope usage in MediaSessionManager

## Description
This is updating the changes that were merged into the `release/7.54` in #1620 to drop the use of GlobalScope like we do in #1619 (which was not on the release branch). In other words, this is basically updating #1620 to what it would have looked like if the removal of GlobalScope changes had been on the release branch.

> [!NOTE]
> This is tagetting the merge branch, so it would be good to merge [that PR](https://github.com/Automattic/pocket-casts-android/pull/1622) first.

## Testing Instructions
 Verify that headphone controls (play, pause, skip forward/back, add bookmark) still work as expected.

## Checklist
